### PR TITLE
[Azure App Configuration] Add purge protection in update store parameter

### PR DESCRIPTION
**Pull request by x9xu**
This PR adds a property that was missed in this PR https://github.com/Azure/azure-rest-api-specs/pull/17444 to add the 2021-10-01-preview api version.

### CODE REVIEW

File: specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2021-10-01-preview/appconfiguration.json
Comment (ruowan) [inline specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2021-10-01-preview/appconfiguration.json:1440]: Is this property outside 'properties' level?

**Comment by jimmyca15** [comment]
@ruowan Can you hold off on merging this for just a bit. We want to get a complete test pass with all changes with a privately generated SDK before merge. 

We'll ping when good to go.

**Comment by jimmyca15** [comment]
@ruowan we have verified the generated client. This can be merged. Thanks.

**Review by ruowan — APPROVED** [review]

**Pull request merged**
